diff.py

```diff
@@ -556,10 +572,14 @@ def table(
 class AnsiFormatter(Formatter):
     column_width: int
 
-    def apply(self, line: str, f: Format) -> str:
-        if f == Format.NONE:
-            return line
-        return f"{f.ansi_code}{line}{Style.RESET_ALL}"
+    def apply(self, text: Text) -> str:
+        output = ""
+        for chunk, f in text.segments:
+            if f == Format.NONE:
+                output += chunk
+            else:
+                output += f"{f.ansi_code}{chunk}{Style.RESET_ALL}"
+        return output
```

Comment:
I think it's preferable to make `output` a `List[str]` and `return "".join(output)` to avoid theoretical O(n^2) performance from O(n) string concatenation... though it's not a big deal since CPython in practice optimizes `x += y` where `x` has refcount = 1.